packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -1656,17 +1764,33 @@ export class Combobox
   }
 
   private renderListBoxOptions(): JsxNode {
-    return this.filteredItems.map((item) => (
-      <li
-        ariaLabel={item.label}
-        ariaSelected={item.selected}
-        id={item.guid ? `${itemUidPrefix}${item.guid}` : null}
-        role="option"
-        tabIndex="-1"
-      >
-        {item.heading || item.textLabel}
-      </li>
-    ));
+    const selectAllComboboxItem = this.selectAllEnabled &&
+      this.selectionMode !== "single" &&
+      this.selectionMode !== "single-persist" && (
+        <calcite-combobox-item
+          id={`${this.guid}-select-all-enabled`}
+          label={this.messages.selectAll}
+          tabIndex="-1"
+          value="select-all"
+        />
+      );
+
+    const selectAllOptionAndFilteredItemsList = [selectAllComboboxItem, ...this.filteredItems];
+
+    return selectAllOptionAndFilteredItemsList.map((item: any) => {
```

Comment:
Can you avoid typing this as any?

Reply:
Can you type this correctly instead of `any`? Maybe just map the filteredItems first and then call `unshift` on the array to add the `selectAll` item to the top of it?

Reply:
Something like this:

```tsx
return [
  selectAllComboboxItem,
  ...this.filteredItems.map((item) => (
    <li
      ariaLabel={item.label}
      ariaSelected={item.selected}
      class={CSS.selectAll}
      id={item.guid ? `${itemUidPrefix}${item.guid}` : null}
      role="option"
      tabIndex="-1"
    >
      {item.heading || item.textLabel}
    </li>
  )),
];
```